### PR TITLE
Clean up Thrift SSL code

### PR DESCRIPTION
**Pull request by ctubbsii**
* Remove unneeded ProtocolOverridingSSLSocketFactory that was used to support older JDK versions that we no longer support
* Clean up SslConnectionParams class to remove a workaround for a bug that was patched upstream in Thrift
* Use Thrift's provided TSSLTransportFactory.getClientSocket method that allows us to provide custom TSSLTransportParameters and delete the private methods copied from Thrift 0.9.1 that were used to implement this ourselves (methods which were not kept up to date with the improvements in upstream Thrift)
* Rename SslConnectionParams.getTTransportParams to .getTSSLTransportParameters, because the old name for the getter method did not reflect the actual type that it returned, and the old name did not make it clear it was returning a parameters object specifically for SSL / TLS

### CODE REVIEW

**Comment by ctubbsii** [comment]
All the SSL tests pass with these changes.

I did see `org.apache.accumulo.core.util.CompletableFutureUtilTest.testIterateUntil` fail. It seems unrelated, but I created #3790 for that.

File: core/src/main/java/org/apache/accumulo/core/rpc/SslConnectionParams.java [hunk 1]
```diff
@@ -222,26 +222,12 @@ public String getTrustStoreType() {
...
     if (useJsse) {
-      throw new IllegalStateException("Cannot get TTransportParams for JSEE configuration.");
-    }
-
-    TSSLTransportParameters params;
-    if (cipherSuites != null) {
-      params = new TSSLTransportParameters(clientProtocol, cipherSuites);
-    } else {
-      params = new TSSLTransportParametersHack(clientProtocol);
+      throw new IllegalStateException("Cannot get TTransportParams for JSSE configuration.");
```
Comment (EdColeman) [inline core/src/main/java/org/apache/accumulo/core/rpc/SslConnectionParams.java:227]:
```suggestion
      throw new IllegalStateException("Cannot get TSSLTransportParameters for JSSE configuration.");
```

**Review by EdColeman — APPROVED** [review]
Minor suggestion in exception message that you can ignore.

**Comment by ctubbsii** [comment]
> Minor suggestion in exception message that you can ignore.

Good catch. Thanks!

**Review by EdColeman — APPROVED** [review]

**Pull request merged**